src/Stryker.Core/Stryker.Core.UnitTest/Mutators/BlockMutatorTests.cs

```diff
@@ -0,0 +1,237 @@
+
+        if (value == 0)
+        {
+            this.alsoThisMustBeInitialized = true;
+        }
+        else
+        {
+            this.alsoThisMustBeInitialized = false;
+        }
+    }
+}";
+
+            GetMutations(source).ShouldBeEmpty();
+        }
+
+        [Fact]
+        private void ShouldMutateLocalFunctionsInStructConstructors()
+        {
+            var source = @"
+struct Program
+{
+    int mustBeInitialized;
+
+    Program(int value)
+    {
+        int CalculateValue()
+        {
+            int value;
+            value = 42; // Try to fool with this assignment
+            return value;
+        }
+
+        this.mustBeInitialized = CalculateValue();
+    }
+}";
+
+            GetMutations(source).Count().ShouldBe(
+                1,
+                "Should mutate the local function and only the local function");
+        }
+
+        [Fact]
+        private void ShouldMutateStructConstructorNonAssignmentsAtRoot()
+        {
+            var source = @"
+struct Program
+{
+    Program(int value)
+    {
+        throw new Exception();
+    }
+}";
+
+            GetMutations(source).Count().ShouldBe(1);
+        }
+
+        [Fact]
+        private void ShouldMutateStructConstructorNonAssignmentChild()
+        {
+            var source = @"
+struct Program
+{
+    int mustBeInitialized;
+
+    Program(int value)
+    {
+        if (value == 0)
+        {
+            throw new Exception();
+        }
+
+        this.mustBeInitialized = value;
+    }
+}";
+
+            GetMutations(source).Count().ShouldBe(1);
+        }
+
+        [Fact]
+        private void ShouldMutateNonVoidReturnsAsDefaultWhenContainsReturns()
+        {
+            var source = @"
+class Program
+{
+    int Method(bool input)
+    {
+        if (input)
+        {
+            input = false;
+        }
+
+        if (input)
+        {
+            return 0;
+        }
+
+        return 1;
+    }
+}";
+
+            var mutations = GetMutations(source).ToList();
+            mutations.Count.ShouldBe(3);
+
+            AssertDefaultReturn(mutations[0]); // Whole method
+            mutations[1].ReplacementNode.ChildNodes().ShouldBeEmpty(); // First if
+            AssertDefaultReturn(mutations[2]); // Second if
+        }
+
+        [Fact]
+        private void ShouldMutateVoidReturnsAsEmptyInMethod()
+        {
+            var source = @"
+class Program
+{
+    void Method(bool input)
+    {
+        if (input)
+        {
+            return;
+        }
+    }
+}";
+
+            GetMutations(source).ShouldAllBe(mutation => !mutation.ReplacementNode.ChildNodes().Any());
+        }
+
+        [Fact]
+        private void ShouldMutateVoidReturnsAsEmptyInLocalFunction()
+        {
+            var source = @"
+class Program
+{
+    int Method(bool input)
+    {
+        void LocalFunction()
+        {
+            return;
+        }
+
+        return 42;
+    }
+}";
+
+            var mutation = GetMutations(source)
```

Comment:
Maybe also assert the number of mutations before filtering them in the where?

Reply:
I feel like it would unnecessarily cause the test to break when other aspects of the mutator change. I'm testing something very specific here, and don't think this test should care about the rest of the possible mutations.

Reply:
Fair enough